src/databricks/labs/lakebridge/connections/credential_manager.py

```diff
@@ -32,18 +38,53 @@ def get_secret(self, key: str) -> str:
             return key
 
 
-class DatabricksSecretProvider:
+class DatabricksSecretProvider(SecretProvider):
+    def __init__(self, ws: WorkspaceClient):
+        self._ws = ws
+
+    def get_databricks_secret(self, scope: str, key: str) -> str:
+        return self.get_secret(f"{scope}/{key}")
+
     def get_secret(self, key: str) -> str:
-        raise NotImplementedError("Databricks secret vault not implemented")
+        """Get the secret value given a secret scope & secret key.
+
+        :param key: key in the format 'scope/secret'
+        :return: The decoded UTF-8 secret value.
+
+        Raises:
+          NotFound: The secret could not be found.
+          UnicodeDecodeError: The secret value was not Base64-encoded UTF-8.
+        """
+        key_parts = key.split(sep="/")
+        assert len(key_parts) == 2, "Secret name must be in the format 'scope/secret'"
+        scope, key_only = key_parts[0], key_parts[1]
```

Comment:
Outside tests I'm pretty sure we limit assertions to things that are demonstrably true but necessary for the type-checker to properly understand the situation. We shouldn't rely on them for validation. (Looking around I see a few places where this is being done incorrectly, but we shouldn't make it worse.)

That leads to something like this:
```suggestion
        match key.split(sep="/", limit=3):
            case scope, key_only:
                scope = scope
                key_only = key_only
            case _:
                msg = f"Secret key must be in the format 'scope/secret': {key}"
                raise ValueError(msg)
```